WhateverGreen/kern_igfx_i2c_aux.cpp

```diff
@@ -0,0 +1,187 @@
+//
+//  kern_igfx_i2c_aux.cpp
+//  WhateverGreen
+//
+//  Created by FireWolf on 9/21/20.
+//  Copyright © 2020 vit9696. All rights reserved.
+//
+
+#include <Headers/kern_util.hpp>
+#include "kern_igfx.hpp"
+
+void IGFX::AdvancedI2COverAUXSupport::init() {
+	// We only need to patch the framebuffer driver
+	requiresPatchingGraphics = false;
+	requiresPatchingFramebuffer = true;
+}
+
+void IGFX::AdvancedI2COverAUXSupport::processKernel(KernelPatcher &patcher, DeviceInfo *info) {
+	// Enable the advanced I2C-over-AUX support if LSPCON is enabled
+	enabled = callbackIGFX->modLSPCONDriverSupport.enabled;
+	
+	// Or if verbose output in transactions is enabled
+	if (checkKernelArgument("-igfxi2cdbg")) {
+		enabled = true;
+		verbose = true;
+	}
+}
+
+void IGFX::AdvancedI2COverAUXSupport::processFramebufferKext(KernelPatcher &patcher, size_t index, mach_vm_address_t address, size_t size) {
+	auto roa = patcher.solveSymbol(index, "__ZN31AppleIntelFramebufferController14ReadI2COverAUXEP21AppleIntelFramebufferP21AppleIntelDisplayPathjtPhbh", address, size);
```

Comment:
Will not it be easier/more readable to use routeMultiple here and in similar places above?